src/main/java/org/dependencytrack/model/Component.java

```diff
@@ -243,7 +243,7 @@ public enum FetchGroup {
 
     @Persistent(defaultFetchGroup = "true")
     @Index(name = "COMPONENT_PURL_IDX")
-    @Size(max = 255)
+    @Column(name = "PURL", jdbcType = "VARCHAR", length = 1024)
```

Comment:
`@Size` should stay as it has a [different purpose](https://beanvalidation.org/) than `@Column(..., length = ...)`.

```suggestion
    @Size(max = 1024)
    @Column(name = "PURL", jdbcType = "VARCHAR", length = 1024)
```